### PR TITLE
Upsample enh

### PR DESCRIPTION
**Pull request by rportilla-databricks**
new PR for upsample capabilities

### CODE REVIEW

File: python/tempo/tsdf.py [hunk 1]
```diff
@@ -429,23 +429,23 @@ def withRangeStats(self, type='range', colsToSummarize=[], rangeBackWindowSecs=1
   def write(self, spark, tabName, optimizationCols = None):
     tio.write(self, spark, tabName, optimizationCols)
 
-  def resample(self, freq, func=None, metricCols = None, prefix=None):
+  def resample(self, freq, func=None, metricCols = None, prefix=None, fill = None):
     """
     function to upsample based on frequency and aggregate function similar to pandas
     :param freq: frequency for upsample - valid inputs are "hr", "min", "sec" corresponding to hour, minute, or second
     :param func: function used to aggregate input
```
Comment (tnixon) [inline python/tempo/tsdf.py:436]: Can we update the documentation on the additional function params not included here:
- metricCols
- prefix
- fill

File: python/tempo/resample.py [hunk 2]
```diff
@@ -24,24 +27,24 @@ def __appendAggKey(tsdf, freq = None):
...
-    return tempo.TSDF(df, tsdf.ts_col, partition_cols = tsdf.partitionCols)
+    return (tempo.TSDF(df, tsdf.ts_col, partition_cols = tsdf.partitionCols), parsed_freq[0], freq_dict[parsed_freq[1]])
 
-def aggregate(tsdf, freq, func, metricCols = None, prefix = None):
+def aggregate(tsdf, freq, func, metricCols = None, prefix = None, fill = None):
     """
     aggregate a data frame by a coarser timestamp than the initial TSDF ts_col
     :param tsdf: input TSDF object
     :param func: aggregate function
     :param metricCols: columns used for aggregates
```
Comment (tnixon) [inline python/tempo/resample.py:44]: Can we document additional params:
- prefix
- fill

**Review by tnixon — CHANGES_REQUESTED** [review]
A couple minor updates requested - see below.

File: python/tempo/resample.py (hunk 2 above)
Comment (rportilla-databricks) [reply python/tempo/resample.py:44]: Thanks

File: python/tempo/tsdf.py (hunk 1 above)
Comment (rportilla-databricks) [reply python/tempo/tsdf.py:436]: Thanks. Done.

**Review by tnixon — APPROVED** [review]
Looks good!

**Pull request merged**